utils/src/treev2/quickPickWizard/compatibility/CompatibilityRecursiveQuickPickStep.ts

```diff
@@ -0,0 +1,120 @@
+            // check if the last picked item is a create callback
+            if (typeof selected.data === 'function') {
+                // If the last node is a function, pop it off the list and execute it
+                const callback = selected.data as unknown as types.CreateCallback<TNode>;
+
+                // context passed to callback must have the same `ui` as the wizardContext
+                // to prevent the wizard from being cancelled unexpectedly
+                const createdPick = await callback?.(wizardContext);
+
+                if (createdPick) {
+                    return createdPick;
+                }
+
+                throw new UserCancelledError();
+            }
+
+            return selected.data;
+        }
+    }
+
+    public async getSubWizard(wizardContext: TContext): Promise<types.IWizardOptions<TContext> | undefined> {
+        const lastPickedItem = getLastNode(wizardContext);
+
+        if (!lastPickedItem) {
+            // Something went wrong, no node was chosen
+            throw new Error('No node was set after prompt step.');
+        }
+
+        if (super.isDirectPick(lastPickedItem)) {
+            // The last picked node matches the expected filter
+            // No need to continue prompting
+            return undefined;
+        } else {
+            // Need to keep going because the last picked node is not a match
+            return {
+                hideStepCount: true,
+                promptSteps: [
+                    new CompatibilityRecursiveQuickPickStep(this.treeDataProvider, {
+                        ...this.pickOptions,
+                        skipIfOne: !lastPickedItem.quickPickOptions.createChild,
+                        create: lastPickedItem.quickPickOptions.createChild,
+                    })
+                ],
+            };
+        }
+    }
+
+    protected override async getPicks(wizardContext: TContext): Promise<types.IAzureQuickPickItem<TNode>[]> {
+        const lastPickedItem: TNode | undefined = getLastNode(wizardContext);
+
+        // TODO: if `lastPickedItem` is an `AzExtParentTreeItem`, should we clear its cache?
+        const children = (await this.treeDataProvider.getChildren(lastPickedItem)) || [];
+
+        const directChoices = children.filter(c => this.isDirectPick(c));
+        const indirectChoices = children.filter(c => this.isIndirectPick(c));
+
+        let promptChoices: (TNode | types.CreateCallback)[] = [];
+
+        if (directChoices.length === 0) {
+            if (indirectChoices.length === 0 && !this.pickOptions.create) {
+                throw new NoResourceFoundError();
+            } else {
+                promptChoices = indirectChoices;
+            }
+        } else {
+            promptChoices = directChoices;
+        }
+
+        const picks: types.IAzureQuickPickItem<TNode | types.CreateCallback>[] = [];
+        for (const choice of promptChoices) {
+            picks.push(await this.getQuickPickItem(choice as TNode));
```

Comment:
This function is nearly identical to `GenericQuickPickStep.getPicks`. Instead of making `getQuickPickItem` `protected`, and duplicating this logic, can you instead do `super.getPicks()` and then push the create pick? Same as what `QuickPickWithCreateStep.ts` does.

Reply:
I had to override it because we currently [throw a `NoResourceFoundError` when there are no picks](https://github.com/microsoft/vscode-azuretools/blob/alex/tip-changes3/utils/src/treev2/quickPickWizard/GenericQuickPickStep.ts#L79).

```ts
if (directChoices.length === 0) {
    if (indirectChoices.length === 0) {
        throw new NoResourceFoundError();
    } else {
        promptChoices = indirectChoices;
    }
} else {
    promptChoices = directChoices;
}
```

In the override method I changed it to

```ts
if (directChoices.length === 0) {
    if (indirectChoices.length === 0 && !this.pickOptions.create) {
        throw new NoResourceFoundError();
    } else {
        promptChoices = indirectChoices;
    }
} else {
    promptChoices = directChoices;
}
```

Reply:
Could we catch and eat that `NoResourceFoundError` instead, if `this.pickOptions.create` is set? It'd be a bit messy but IMO better than duplicate (or near-duplicate) code.

Reply:
Good idea

Reply:
I think this goes without saying, but some comments to explain why it's catching and eating that error would be good too. I think the standard `QuickPickWithCreateStep` needs this same logic; however I don't feel too badly about duplicating a couple lines vs. a whole function.

Reply:
Ok, added the logic to `QuickPickWithCreateStep`. FYI I haven't tested/used `QuickPickWithCreateStep` yet, so there may need to be changes once I get to adding that to the experience builder API.